### PR TITLE
Automated cherry pick of #90248: Add back anti-affinity to kube-dns pods.

### PR DESCRIPTION
**Pull request by prameshj**
Cherry pick of #90248 on release-1.17.

#90248: Add back anti-affinity to kube-dns pods.

For details on the cherry pick process, see the [cherry pick requests](https://git.k8s.io/community/contributors/devel/sig-release/cherry-picks.md) page.

### CODE REVIEW

**Comment by alculquicondor** [comment]
This has been verified to not cause performance regressions in 1.17

/lgtm

from SIG scheduling

**Review by MrHohn — APPROVED** [review]
/approve

**Comment by tpepper** [comment]
/kind feature
from parent PR.

@prameshj generally we don't backport features, can you confirm with the code OWNERS whether this is something to cherry pick to 1.18 and 1.17?

**Comment by prameshj** [comment]
Since this is a change to spread kube-dns pods across nodes and thereby help prevent single point of failure of DNS, I was hoping this could be considered as a backport candidate.
 cc @MrHohn @bowei

**Comment by tpepper** [comment]
Discussed in SIG Release today:  We're ok with the reassessment of this being a bug.  Since we're past the cherry pick deadline for this week and have some CI instability on 1.18/1.17, we plan to cherry pick approve for merge after Wednesday's expected May patch release is cut.  This means we will have time to see if we hit any scale issues on 1.19/1.18/1.17 and do any additional fixups ahead of June's patch releases.

**Comment by prameshj** [comment]
> Discussed in SIG Release today: We're ok with the reassessment of this being a bug. Since we're past the cherry pick deadline for this week and have some CI instability on 1.18/1.17, we plan to cherry pick approve for merge after Wednesday's expected May patch release is cut. This means we will have time to see if we hit any scale issues on 1.19/1.18/1.17 and do any additional fixups ahead of June's patch releases.

Thanks Tim and Stephen! Also want to note that we are only cherrypicking only as far back as 1.17 since most scheduling improvements were done in that version. Older versions might have performance regression with this change.

**Comment by tpepper** [comment]
> Discussed in SIG Release today: We're ok with the reassessment of this being a bug. Since we're past the cherry pick deadline for this week and have some CI instability on 1.18/1.17, we plan to cherry pick approve for merge after Wednesday's expected May patch release is cut. This means we will have time to see if we hit any scale issues on 1.19/1.18/1.17 and do any additional fixups ahead of June's patch releases.

as per ^^
/kind bug

**Pull request merged**
